### ISSUE TITLE
Mapper fails to generate

### ISSUE THREAD

**Issue by dustincatap** [smotastic/smartstruct#35]
`Mapper got case insensitive fields and contains fields: data and data. If you use a case-sensitive mapper, make sure the fields are unique in a case insensitive way.`

I'm using the latest version. This was still working v1.2.3+1

**Comment by smotastic** [comment]
Hi @dustincatap
Thank you for your report.

I think the problem might be connected to #30.
Here every field (and it's superclasses fields) are taken into account for the creation of the mapper.

I could reproduce your bug if I have a class with a superclass which are sharing a field with the same name.
```dart
class Foo extends SuperFoo {
  final String fooBar;

  Foo(this.fooBar) : super(fooBar);
}

class SuperFoo {
  final String fooBar;
  SuperFoo(this.fooBar);
}

class Bar {
  final String fooBar;

  Bar(this.fooBar);
}

@Mapper()
abstract class FooBarMapper {
  Foo fromBar(Bar bar);
}
```

I think the error message is misleading here, as it has nothing to do with the case sensitivity of the mapper, as it is just a duplicated key. 
Also if a class and its superclass share a field, precedence should apply for the fields in the subclass.

Is this also your case? If not can you please provide a minimalistic example of your error showcasing the bug?

**Comment by dustincatap** [comment]
Does it have to do with interfaces? The mapper that gives this error maps the following classes (`MyContract -> MyEntity`):

Source:

```dart
abstract class DataContract {
  String get id;
}

class MyContract implements DataContract {
  MyContract(this.id);

  @override
  final String id;
}
```

Target:

```dart
class MyEntity {
  MyEntity(this.id);

  final String id;
}
```

**Comment by smotastic** [comment]
Yes, this seems to be the problem.
I didn't think about this, but now that I see it it's a pretty obvious use case.

The fix should be to just filter out the abstract getters. Or abstract fields in general.
I'll get to it

**Comment by smotastic** [comment]
Hey @dustincatap 
I just published a hotfix for this issue (1.2.4+1).
I also extended my inheritance example by a few more use cases, so hopefully, it should work now.
For your use case, I also added a specific example, see here https://github.com/smotastic/smartstruct/blob/develop/example/lib/inheritance/inheritance_35.dart

I hope this resolves your issue with the newest version of smartstruct :)
If there are any other problems, feel free to open a new Issue
